test/CMakeLists.txt

```diff
@@ -1,144 +1,64 @@
-cmake_minimum_required(VERSION 3.13)
-project(godot-cpp-test)
-
-set(GODOT_GDEXTENSION_DIR ../gdextension/ CACHE STRING "Path to GDExtension interface header directory")
-set(CPP_BINDINGS_PATH ../ CACHE STRING "Path to C++ bindings")
-
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-	set(TARGET_PATH x11)
-elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-	set(TARGET_PATH win64)
-elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-	set(TARGET_PATH macos)
-else()
-	message(FATAL_ERROR "Not implemented support for ${CMAKE_SYSTEM_NAME}")
-endif()
-
-# Change the output directory to the bin directory
-set(BUILD_PATH ${CMAKE_SOURCE_DIR}/bin/${TARGET_PATH})
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${BUILD_PATH}")
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${BUILD_PATH}")
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${BUILD_PATH}")
-SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG "${BUILD_PATH}")
-SET(CMAKE_RUNTIME_OUTPUT_DIRECTORY_RELEASE "${BUILD_PATH}")
-SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY_DEBUG "${BUILD_PATH}")
-SET(CMAKE_LIBRARY_OUTPUT_DIRECTORY_RELEASE "${BUILD_PATH}")
-SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_DEBUG "${BUILD_PATH}")
-SET(CMAKE_ARCHIVE_OUTPUT_DIRECTORY_RELEASE "${BUILD_PATH}")
-
-# Set the c++ standard to c++17
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
-set(CMAKE_CXX_EXTENSIONS OFF)
-
-set(GODOT_COMPILE_FLAGS )
-set(GODOT_LINKER_FLAGS )
-
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
-	# using Visual Studio C++
-	set(GODOT_COMPILE_FLAGS "${GODOT_COMPILE_FLAGS} /WX") # /GF /MP
-	set(GODOT_COMPILE_FLAGS "${GODOT_COMPILE_FLAGS} /DTYPED_METHOD_BIND")
-	set(GODOT_COMPILE_FLAGS "${GODOT_COMPILE_FLAGS} /utf-8")
-
-	if(CMAKE_BUILD_TYPE MATCHES Debug)
-		set(GODOT_COMPILE_FLAGS "${GODOT_COMPILE_FLAGS} /MDd") # /Od /RTC1 /Zi
-	else()
-		set(GODOT_COMPILE_FLAGS "${GODOT_COMPILE_FLAGS} /MD /O2") # /Oy /GL /Gy
-		STRING(REGEX REPLACE "/RTC(su|[1su])" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-		string(REPLACE "/RTC1" "" CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
-	endif(CMAKE_BUILD_TYPE MATCHES Debug)
-
-	# Disable conversion warning, truncation, unreferenced var, signed mismatch
-	set(GODOT_COMPILE_FLAGS "${GODOT_COMPILE_FLAGS} /wd4244 /wd4305 /wd4101 /wd4018 /wd4267")
-
-	add_definitions(-DNOMINMAX)
-
-	# Unkomment for warning level 4
-	#if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
-	#	string(REGEX REPLACE "/W[0-4]" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-	#endif()
-
-else()
+# Testing Extension
+# This is only linked to the template_release config.
+# so it requires the template_release version of godot to run.
```

Comment:
I see what I've gone and done, I had it in my head that the default was template_release, I was following the options from the testing in CI. I'll change the defaults to template_debug.

There are many ways to configure a build, for consumers it will likely be some combination of things like:
```cmake
# They can change the godot options before adding it as a subproject, or on the command line, 
set( GODOT_PRECISION "double" )
add_subdirectory( godot-cpp )

# have their own options to use
set( EDITOR ON )

# They can use IF statements and set it at configure time
IF( EDITOR )
  set( GODOT_LIB "godot-cpp::editor" )
else()
  set( GODOT_LIB "godor-cpp::template_debug" )
endif()
target_link_libraries( my_extension
  PRIVATE
    $GODOT_LIB
)
#or if they are using generator expressions this will choose at build time.
target_link_libraries( my_extension
  PRIVATE
    $<IF:$<BOOL:${EDITOR}>,godot-cpp::editor, $<$<Release>:godot-cpp::template_release>$<$<Debug>:godot-cpp::template_debug>>
)
```

Reply:
I have make it so that just the basename "godot-cpp" by itself isnt associated with a target, this is because when I went to build a project that relies on any of the target aliases(editor, template_etc) it would always build whatever was added to the ALL targets, aka the default is to build all. It was wasting time building ALL before it built the alias. I didnt manage to figure that out and decided to drop ALL as a useful concept. Thats why there is no default, and linking to a target alias ie "godot-cpp::editor" is the way it's done.

Reply:
That sounds good!

We have `template_debug` as the default with SCons because it's compatible with both editor and debug export templates, so folks can make just the one build for active development. (But when releasing their GDExtension, they may want to provide their users with multiple builds.)